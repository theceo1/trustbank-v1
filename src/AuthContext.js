import React, { createContext, useContext, useState } from 'react';

const AuthContext = createContext();

export const AuthProvider = ({ children }) => {
  const [user, setUser] = useState(null);

  const login = async ({ email, password }) => {
    // Simulate an API call for login
    if (email === 'testing4@test.com' && password === 'password') {
      setUser({ email });
    } else {
      throw new Error('Invalid credentials');
    }
  };

  const logout = () => {
    setUser(null);
  };

  return (
    <AuthContext.Provider value={{ user, login, logout }}>
      {children}
    </AuthContext.Provider>
  );
};

export const useAuth = () => {
  return useContext(AuthContext);
};
