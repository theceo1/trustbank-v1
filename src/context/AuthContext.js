// src/context/AuthContext.js
import React, { createContext, useState, useContext, useEffect } from 'react';
import { useRouter } from 'next/router';

const AuthContext = createContext();

export const AuthProvider = ({ children }) => {
  const [user, setUser] = useState(null);
  const router = useRouter();

  useEffect(() => {
    // Load user from local storage or API
    const loggedInUser = localStorage.getItem('user');
    if (loggedInUser) {
      setUser(JSON.parse(loggedInUser));
    }
  }, []);

  const login = (userData) => {
    setUser(userData);
    localStorage.setItem('user', JSON.stringify(userData));
    router.push('/dashboard');
  };

  const logout = () => {
    setUser(null);
    localStorage.removeItem('user');
    router.push('/login');
  };

  return (
    <AuthContext.Provider value={{ user, login, logout }}>
      {children}
    </AuthContext.Provider>
  );
};

export const useAuth = () => useContext(AuthContext);
