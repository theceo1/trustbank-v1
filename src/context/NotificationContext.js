import React, { createContext, useContext, useState, useCallback } from 'react';
import { v4 as uuidv4 } from 'uuid';

const NotificationContext = createContext();

export const useNotification = () => useContext(NotificationContext);

export const NotificationProvider = ({ children }) => {
  const [notifications, setNotifications] = useState([]);

  const addNotification = useCallback((type, message) => {
    const id = uuidv4();
    setNotifications((prev) => [...prev, { id, type, message }]);
    setTimeout(() => {
      setNotifications((prev) => prev.filter((notification) => notification.id !== id));
    }, 5000);
  }, []);

  return (
    <NotificationContext.Provider value={{ notifications, addNotification }}>
      {children}
    </NotificationContext.Provider>
  );
};
