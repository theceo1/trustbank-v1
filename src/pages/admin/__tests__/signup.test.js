import React from 'react';
import { render } from '@testing-library/react';
import Signup from '../signup';
import { AuthProvider } from '@/context/AuthContext';

const renderWithAuthProvider = (ui, { providerProps, ...renderOptions }) => {
  return render(
    <AuthProvider {...providerProps}>{ui}</AuthProvider>,
    renderOptions
  );
};

test('renders Signup component', () => {
  const { asFragment } = renderWithAuthProvider(<Signup />, {
    providerProps: { value: { user: null } },
  });
  expect(asFragment()).toMatchSnapshot();
});
