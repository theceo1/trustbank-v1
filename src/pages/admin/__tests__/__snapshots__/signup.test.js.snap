// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Signup component 1`] = `
<DocumentFragment>
  <div
    class="flex h-screen items-center justify-center bg-gray-100"
  >
    <div
      class="w-full max-w-md"
    >
      <div
        class="bg-white p-8 rounded-lg shadow-lg"
      >
        <div
          class="text-center mb-4"
        >
          <img
            alt="Logo"
            data-nimg="1"
            decoding="async"
            height="50"
            loading="lazy"
            src="/images/logo/logo.svg"
            style="color: transparent;"
            width="100"
          />
        </div>
        <h2
          class="text-2xl font-bold mb-4 text-center"
        >
          Sign Up to TailAdmin
        </h2>
        <form>
          <div
            class="mb-4"
          >
            <label
              class="block text-gray-700"
            >
              Email
            </label>
            <input
              class="w-full p-2 border border-gray-300 rounded mt-2"
              placeholder="Enter your email"
              type="email"
              value=""
            />
          </div>
          <div
            class="mb-4"
          >
            <label
              class="block text-gray-700"
            >
              Password
            </label>
            <input
              class="w-full p-2 border border-gray-300 rounded mt-2"
              placeholder="Enter your password"
              type="password"
              value=""
            />
          </div>
          <div
            class="mb-4"
          >
            <label
              class="block text-gray-700"
            >
              Re-type Password
            </label>
            <input
              class="w-full p-2 border border-gray-300 rounded mt-2"
              placeholder="Re-enter your password"
              type="password"
              value=""
            />
          </div>
          <button
            class="w-full bg-blue-500 text-white py-2 rounded hover:bg-blue-600"
            type="submit"
          >
            Create account
          </button>
        </form>
        <div
          class="mt-4 text-center"
        >
          <p
            class="text-gray-700"
          >
            Already have an account? 
            <a
              class="text-blue-500"
              href="/admin/signin"
            >
              Sign in
            </a>
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
